feature-pitches/002-file-based-workflows/002-file-based-workflows.md

```diff
@@ -0,0 +1,106 @@
+
+CLI tools require the user to maintain a mental model of what’s happening with every command they run. This additional layer of abstraction is the reason why we only expect advanced users to use CLIs at all. For example, a person who understands git well gets the job done quickly with CLI commands, but someone new to git will prefer to use their IDE’s GUI or the GitHub web or desktop app.
+
+As long as Renku workflows are only available on the CLI, only the most advanced users will use them. But Renku’s mission is to make reproducible research *easy* and for *everyone*, not only the technically advanced! So, our first motivating question is *what can we do to make workflows accessible to a broader audience?*
+
+Additionally, there is a common challenge we run into when introducing Renku workflows to users who *do* want to use workflows: it is cumbersome/just-not-tenable to create and develop (modify) workflows of more than a few steps. Many workflow users have 10+ steps in their workflows [[source](https://sdsc.atlassian.net/wiki/spaces/RENKU/pages/2246541341/Reproducibility+on+HPC)]. Entering 10+ steps on the command line is tedious to do once, but even more significant is that modifying that workflow- which users do all the time as they develop their pipeline- requires recreating and rewiring steps again and again on the CLI. Some users end up writing a script to call a series of Renku run commands [[ORDES example](https://renkulab.io/projects/renku-stories/digirhythm/files/blob/src/build_workflow.sh)]. Renku should provide a built-in way to create and develop multi-step workflows via a file!
+
+## 🍴 Appetite
+
+We’d like to invest 2 sprints (6 weeks) for making workflows more accessible to a broader audience.
+
+## 🎯 Solution
+
+We’d like to offer the ability to define Renku workflows in a file, where the full workflow is laid out for the user to plainly see and edit.
+
+I’ve done a [survey of popular workflow tools](https://sdsc.atlassian.net/wiki/spaces/RENKU/pages/2310701105/Workflow+Tool+Research), and YAML seems to be a common choice for a workflow file, but we leave the final decision to the team.
+
+<img src="fat-marker-file.png" width="400">
+
+The workflow file names a sequence of steps, and each step names its command, inputs, outputs, parameters, etc. Each step should be named so that it can be referenced for running individually. I imagine that the structure of the workflow step definition in the file would be pretty similar to the output of `renku workflow show`.
+
+As a nice-to-have, the workflow definition file should allow for the specification of workflow annotations and additional metadata.
+
+![renku workflow show](workflow-show.png)
+
+For the scope of this pitch, expressing `renku workflow iterate` in the workflow file definition syntax is _out of scope_, along with any kinds of looping or branching.
```

Comment:
That `iterate` is out of scope is almost by-definition - `renku workflow iterate` is to execute an existing workflow with a series of parameters, it's not something expressed in the workflow itself.